### DISCUSSION TITLE
Per Pixel Confidence Map for Segmentations

### DISCUSSION THREAD

**Discussion by bhemp** [facebookresearch/detectron2#5362] [Q&A]
Hi there,

I'm working on a medical image segmentation project where often the outer boundaries of organs are ambiguous and fuzzy, but the center of the organ is more well defined. Is there any way to produce a confidence score/map for each pixel of the binary segmentation so that I could later remove pixels below a certain confidence threshold?

**Comment by bmaxdk** [comment]
Can you use combination of Monte Carlo Dropout (Detectron2 models do not have dropout layers by default during inference. You will need to add dropout layers manually to your model's architecture if you planning to use.) and Softmax/Sigmoid confidence scores for pixel-wise confidence maps? Since keeping dropout active during inference and averaging over multiple predictions, you can estimate pixel-wise uncertainty, especially around the fuzzy, ambiguous boundaries of organs. Once you have the per-pixel confidence scores, you can apply a threshold to remove pixels that are below a certain confidence level. For post processing, you can refine the segmentation by applying techniques such as dilation, erosion...etc to smooth the segmentation mask and reduce noise from low-confidence pixels.
